src/bluesky/bundlers.py

```diff
@@ -715,46 +678,34 @@ async def _describe_collect(self, collect_object: Flyable):
         """
         await self._ensure_cached(collect_object, collect=True)
 
-        describe_collect = self._describe_collect_cache[collect_object]
-        describe_collect_items = list(self._maybe_format_datakeys_with_stream_name(describe_collect))
-
         local_descriptors: Dict[Any, Dict[FrozenSet[str], ComposeDescriptorBundle]] = {}
 
-        # Check that singly nested stuff should have been pre-declared
-        def is_data_key(obj: Any) -> bool:
-            return isinstance(obj, dict) and {"dtype", "shape", "source"}.issubset(frozenset(obj.keys()))
-
-        assert all(
-            not is_data_key(value) for value in describe_collect.values()
-        ), "Single nested data keys should be pre-decalred"
-
-        # Make sure you can't use identidal data keys in multiple streams
+        # Make sure you can't use identical data keys in multiple streams
         duplicates: Dict[str, DataKey] = defaultdict(dict)
-        for stream, data_keys in describe_collect.items():
-            for key, stuff in data_keys.items():
-                for other_stream, other_data_keys in describe_collect.items():
-                    for other_key, other_stuff in other_data_keys.items():
-                        if stream != other_stream and key == other_key and stuff == other_stuff:
-                            duplicates[stream][key] = stuff
+        for stream, device_map in self._describe_collect_cache.items():
```

Comment:
Maybe `itertools.combinations` here to skip the repeats?

Reply:
Think this is slightly more efficient?